### PR TITLE
First pass at JMS

### PR DESCRIPTION
**Pull request by ggam**
I took the liberty to use "Messaging" in place of "Jakarta Messaging" in a lot of places to reduce verbosity. This is in line with the use of "enterprise beans" that was already there.

Going forward, I think we'll better use generic terms such as "messaging" or "transactions" in these non normative texts to make texts more user friendly.

### CODE REVIEW

File: src/main/jbake/content/jms-concepts001.adoc [hunk 1]
```diff
@@ -126,53 +124,52 @@ might work.
...
 services applications, and many others can make use of messaging.
 
 [[BNCDW]][[how-does-the-jms-api-work-with-the-jakarta-ee-platform]]
 
-How Does the JMS API Work with the Jakarta EE Platform?
+How Does Jakarta Messaging Work with the Jakarta EE Platform?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When the JMS API was first introduced, its most important purpose was to
+When JMS was first introduced, its most important purpose was to
```
Comment (ggam) [inline src/main/jbake/content/jms-concepts001.adoc]: I left this JMS references as they talk about history.

File: src/main/jbake/content/jms-examples.adoc [hunk 2]
```diff
@@ -1,30 +1,30 @@
 type=page
 status=published
-title=Java Message Service Examples
+title=Jakarta Messaging Examples
 next=jms-examples001.html
 prev=jms-concepts006.html
 ~~~~~~
-Java Message Service Examples
+Java Messaging Examples
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples.adoc]: Should be "Jakarta Messaging Examples".

File: src/main/jbake/content/jms-examples.adoc [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
 =============================
 
 [[BNCGV]][[java-message-service-examples]]
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples.adoc]: Ideally any of these tags would be updated as well, e.g., to "jakarta-message-service-examples".

File: src/main/jbake/content/jms-examples.adoc [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
 
-49 Java Message Service Examples
+49 Java Messaging Examples
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples.adoc]: Jakarta Messaging Examples

File: src/main/jbake/content/jms-examples002.adoc [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
-title=Overview of the JMS Examples
+title=Overview of the Jakarta Messaging Examples
 next=jms-examples003.html
 prev=jms-examples001.html
 ~~~~~~
-= Overview of the JMS Examples
+= Overview of the Jakarta Messaging Examples
 
 
 [[BABEFBHJ]][[overview-of-the-jms-examples]]
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples002.adoc]: overview-of-the-jakarta-messaging-examples

File: src/main/jbake/content/jms-examples002.adoc [hunk 6]
```diff
@@ -63,7 +63,7 @@ link:jms-examples005.html#BABEJBHA[Using Shared Durable Subscriptions]
 
 [[sthref202]][[sthref203]]
 
-*Table 49-2 JMS Examples That Show the Use of Jakarta EE Web and EJB
+*Table 49-2 Jakarta Messaging Examples That Show the Use of Jakarta EE Web and EJB
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples002.adoc]: EJB -> Enterprise Beans

File: src/main/jbake/content/jms-examples003.adoc [hunk 7]
```diff
@@ -1,25 +1,25 @@
...
-title=Writing Simple JMS Applications
+title=Writing Simple Jakarta Messaging Applications
 next=jms-examples004.html
 prev=jms-examples002.html
 ~~~~~~
-= Writing Simple JMS Applications
+= Writing Simple Jakarta Messaging Applications
 
 
 [[BNCFA]][[writing-simple-jms-applications]]
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples003.adoc]: writing-simple-jakarta-messaging-applications

File: src/main/jbake/content/jms-examples003.adoc [hunk 8]
```diff
@@ -31,10 +31,10 @@ Delivery]
 
 [[CHDCEFGA]][[overview-of-writing-simple-jms-application]]
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples003.adoc]: overview-of-writing-simple-jakarta-messaging-application

(You get the idea, I'm not going to keep marking these.)

**Comment by ggam** [comment]
All done but the change of tags.

File: src/main/jbake/content/jms-concepts.adoc [hunk 9]
```diff
@@ -1,30 +1,30 @@
...
-:keywords: jms, glassfish
+= Jakarta Messaging Concepts
+:description: Jakarta Messaging allows applications to create, send, and receive messages.
+:keywords: messaging, glassfish
 
 [[BNCDQ]][[java-message-service-concepts]]
 
-48 Java Message Service Concepts
+48 Jakarta Messaging Concepts
 --------------------------------
```
Comment (bshannon) [inline src/main/jbake/content/jms-concepts.adoc]: This underline is too long.

File: src/main/jbake/content/jms-concepts003.adoc [hunk 10]
```diff
@@ -85,24 +85,24 @@ override those specified by annotations.
 
 [[BNCEK]][[jms-connection-factories]]
 
-JMS Connection Factories
-^^^^^^^^^^^^^^^^^^^^^^^^
+Jakarta Messaging Connection Factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (bshannon) [inline src/main/jbake/content/jms-concepts003.adoc]: This underline is one too short.

File: src/main/jbake/content/jms-concepts003.adoc [hunk 11]
```diff
@@ -113,21 +113,21 @@ private static ConnectionFactory connectionFactory;
 
 [[BNCEL]][[jms-destinations]]
 
-JMS Destinations
-^^^^^^^^^^^^^^^^
+Jakarta Messaging Destinations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (bshannon) [inline src/main/jbake/content/jms-concepts003.adoc]: Ditto.

File: src/main/jbake/content/jms-concepts003.adoc [hunk 12]
```diff
@@ -376,8 +376,8 @@ link:#BABJCIGJ[Creating Shared Subscriptions].
 
 [[BNCEQ]][[jms-message-listeners]]
 
-JMS Message Listeners
-^^^^^^^^^^^^^^^^^^^^^
+Jakarta Messaging Message Listeners
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (bshannon) [inline src/main/jbake/content/jms-concepts003.adoc]: Ditto.

File: src/main/jbake/content/jms-concepts003.adoc [hunk 13]
```diff
@@ -419,13 +419,13 @@ and message-driven beans.
 
 [[BNCER]][[jms-message-selectors]]
 
-JMS Message Selectors
-^^^^^^^^^^^^^^^^^^^^^
+Jakarta Messaging Message Selectors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (bshannon) [inline src/main/jbake/content/jms-concepts003.adoc]: Ditto.

File: src/main/jbake/content/jms-concepts005.adoc [hunk 14]
```diff
@@ -243,39 +243,39 @@ and to consume messages.
...
+Using Jakarta Messaging in a Jakarta EE component is in many ways similar to
 using it in an application client. The main differences are the areas of
 resource management and transactions.
 
 [[BNCGO]][[managing-jms-resources-in-web-and-ejb-components]]
 
-Managing JMS Resources in Web and EJB Components
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Managing Jakarta Messaging Resources in Web and Jakarta Enterprise Beans Components
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (bshannon) [inline src/main/jbake/content/jms-concepts005.adoc]: Ditto.

(Did you write a script to fix the underlines?  Only the "^" underlines are wrong.)

File: src/main/jbake/content/jms-examples.adoc [hunk 15]
```diff
@@ -1,30 +1,30 @@
...
-=============================
+Jakarta Messaging Examples
+==========================
 
 [[BNCGV]][[java-message-service-examples]]
 
-49 Java Message Service Examples
---------------------------------
+49 Jakarta Messaging Examples
+--------------------------
```
Comment (bshannon) [inline src/main/jbake/content/jms-examples.adoc]: Underline too short.

File: src/main/jbake/content/jms-concepts005.adoc (hunk 14 above)
Comment (ggam) [reply src/main/jbake/content/jms-concepts005.adoc]: Let's say I'm using visual recognition :). Perhaps I should try to write some scripts for this.

I think I have now fixed all. Thanks for the review.

Comment (bshannon) [reply src/main/jbake/content/jms-concepts005.adoc]: I guess it's time for a Maven plugin...  :-)

Comment (ggam) [reply src/main/jbake/content/jms-concepts005.adoc]: Hopefully we can eventually get rid those underlines...

Comment (bshannon) [reply src/main/jbake/content/jms-concepts005.adoc]: Yes, there's a different style for AsciiDoc section titles that doesn't use underlines, but...
1. The Maven plugin that generates the Table of Contents needs to be updated to understand it.
2. The title style has subtle implications on some other AsciiDoc syntax that needs to be considered.

Comment (ggam) [reply src/main/jbake/content/jms-concepts005.adoc]: I will give it a try now we have completed the very first step. Btw thanks a lot for you thorough reviews!

**Pull request merged**